### PR TITLE
fix(start_planner): not run other modules when back pull out

### PR DESCRIPTION
**Pull request by kosuke55**
## Description

```
    setIsSimultaneousExecutableAsApprovedModule(false);
    setIsSimultaneousExecutableAsCandidateModule(false);
```

when backward driving of pull out.

this prevents LC from being approved when back and generating invalid path

[Screencast from 2023年06月22日 11時24分17秒.webm](https://github.com/autowarefoundation/autoware.universe/assets/39142679/57612182-2e0c-48ed-98f5-cd8c947122a3)

## Related links

[tier4 internal ticket](https://tier4.atlassian.net/browse/RT1-2852)

## Tests performed

psim

https://github.com/autowarefoundation/autoware.universe/assets/39142679/1a5232e6-f99d-4d67-841a-44ba3cfbb0de

## Interface changes

none
## Effects on system behavior

none

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/include/behavior_path_planner/scene_module/start_planner/start_planner_module.hpp [hunk 1]
```diff
@@ -104,6 +104,19 @@ class StartPlannerModule : public SceneModuleInterface
   {
   }
 
+  // set is_simulataneously_executable_ as false when backward driving.
```
Comment (satoshi-ota) [inline planning/behavior_path_planner/include/behavior_path_planner/scene_module/start_planner/start_planner_module.hpp:107]:
```suggestion
  // set is_simultaneously_executable_ as false when backward driving.
```

File: planning/behavior_path_planner/include/behavior_path_planner/scene_module/start_planner/start_planner_module.hpp [hunk 2]
```diff
@@ -104,6 +104,19 @@ class StartPlannerModule : public SceneModuleInterface
...
+  // keep initial value to return it after finishing backward driving.
+  bool initial_is_simultaneously_executable_as_approved_module_;
```
Comment (satoshi-ota) [inline planning/behavior_path_planner/include/behavior_path_planner/scene_module/start_planner/start_planner_module.hpp:109]: [imo] `initial_is_*_ is` little bit a unreadable.
`initial_value_simultaneously_executable_as_approved_module_` ?

**Review by satoshi-ota — APPROVED** [review]

**Comment by shmpwk** [comment]
We tested this with real vehicle and works fine.

File: planning/behavior_path_planner/include/behavior_path_planner/scene_module/start_planner/start_planner_module.hpp (hunk 2 above)
Comment (kosuke55) [reply planning/behavior_path_planner/include/behavior_path_planner/scene_module/start_planner/start_planner_module.hpp]: it' better, fixed in 9c215eaa6aa72d5547f40e324cdd5a75cd001f28

**Review by shmpwk — APPROVED** [review]

**Pull request merged**